src/platform/unix/mod.rs

```diff
@@ -649,10 +649,69 @@ fn make_socket_lingering(sockfd: c_int) -> Result<(),UnixError> {
     Ok(())
 }
 
+struct BackingStore {
```

Comment:
In my original design, this played a much bigger role. The `BackingStore` doesn't really do much as is, and could easily be removed. We really rely on the `create_shmem` function.